### PR TITLE
Make *ModelFactory take IEnumerable<X> as method arguments

### PR DESCRIPTION
**Pull request by AlexanderSher**
Some methods are restored manually because they are GA'd

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
+using Azure.Core;
 
 namespace Azure.Communication.Chat
 {
     /// <summary>
     /// Model factory that enables mocking for the Chat library.
     /// </summary>
-    public static class ChatModelFactory
+    [CodeGenType("Azure.Communication.Chat.AzureCommunicationServicesModelFactory")]
+    public static partial class ChatModelFactory
```
Comment (pakrym) [inline sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:15]: Is it easier to change the `title` to `Chat` in autorest.md?

Comment (AlexanderSher) [reply sdk/communication/Azure.Communication.Chat/src/Models/ChatModelFactory.cs:14]: Fixed

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs [hunk 2]
```diff
@@ -1,16 +1,141 @@
...
 using System.Text.Json;
-using Azure.Core;
 using Azure.Messaging.EventGrid.SystemEvents;
 
 namespace Azure.Messaging.EventGrid
 {
 #pragma warning disable CA1054 // URI-like parameters should not be strings
     public static partial class EventGridModelFactory
     {
+        /// <summary> Initializes new instance of MediaJobError class. </summary>
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs:15]: Can we hide all of these with EditorBrowsableState.Never?

Comment (AlexanderSher) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs:15]: Sure

Comment (AlexanderSher) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs:15]: Fixed

**Review by JoshLove-msft — APPROVED** [review]
Event Grid changes look good.

**Pull request merged**